### PR TITLE
Generating certs for TLS and  using them for public endpoints

### PR DESCRIPTION
**Pull request by kantarun**
Ingress certs are generated for each new deployment and are
different from airship default certs.

Certs are generated via cfssl (CloudFare Utility). Airship promenade
uses same tool to generate SSL certs which are bundled in airship
treasuremap sample sites.

No TLS for public endpoints in ucp.
Added TLS for keystone, cinder, nova, neutron, hoirzon public endpoints
in openstack namespace.

Need to change ingress domain suffix from 'svc.cluster.local' to
'cluster.local' so that tls secret reference is added in k8s ingress
definition for public endpoints. This is different from upstream
treasuremap airskiff site reference. Ingress k8s definition is generated
by helm-toolkit _ingress.tpl template and should have domain different
from svc.{cluster_domain_suffix} value to add tls section.

### CODE REVIEW

File: playbooks/roles/airship-deploy-ucp/tasks/main.yml [hunk 1]
```diff
@@ -55,6 +57,17 @@
     - install
     - update_airship_osh_site
 
+- name: Create site network manifests
```
Comment (jgu17) [inline playbooks/roles/airship-deploy-ucp/tasks/main.yml]: why are we separating the network manifest copy from the rest?

File: playbooks/roles/airship-deploy-ucp/defaults/main.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 ---
+ucp_namespace_name: "{{ ucp_namespace | default('ucp') }}"
```
Comment (jgu17) [inline playbooks/roles/airship-deploy-ucp/defaults/main.yml]: namespace variables are already defined in common-vars.yaml.

File: playbooks/roles/airship-deploy-ucp/defaults/main.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
...
+ceph_namespace_name: "{{ ceph_namespace | default('ceph') }}"
```
Comment (jgu17) [inline playbooks/roles/airship-deploy-ucp/defaults/main.yml]: do yo need ceph name space name here?

File: examples/workdir/env/extravars [hunk 4]
```diff
@@ -19,6 +19,12 @@ redeploy_osh_only: false
 #suse_osh_registry_location: "docker.io"
 #suse_osh_image_version: "pike"
 
+# For TLS CA cert: common name used by default
+#suse_tls_cert_CA_common_name: "ca.cluster.local"
+
+# For TLS cert: common name used by default
+#suse_tls_cert_common_name: "server.cluster.local"
```
Comment (jgu17) [inline examples/workdir/env/extravars]: we should allow user defined domain suffix. but that can be in a different commit.

File: playbooks/roles/airship-deploy-ucp/tasks/main.yml (hunk 1 above)
Comment (kantarun) [reply playbooks/roles/airship-deploy-ucp/tasks/main.yml]: As now network file has jinja template code so need to be copied as template which is done later.

File: playbooks/roles/airship-deploy-ucp/defaults/main.yml (hunk 3 above)
Comment (kantarun) [reply playbooks/roles/airship-deploy-ucp/defaults/main.yml]: Just copied from various places. We copy site file so not sure we need it or not ?
Ideally these should be defined in common-vars and referred wherever needed. But somehow agreement is to have them defined in individual role defaults and then it follow boilerplate code pattern and copied to keep it consistent usage.

File: playbooks/roles/airship-deploy-ucp/defaults/main.yml (hunk 2 above)
Comment (kantarun) [reply playbooks/roles/airship-deploy-ucp/defaults/main.yml]: Oddly, only ucp namespace is defined. hmm, may be should define other 2 there as well. Will do that

**Comment by rhafer** [comment]
@kantarun Is this still needed or can we close it for now? You could just reopen if you find time to work on it again.

**Comment by Itxaka** [comment]
Lets close this for now, it can be restored later
